### PR TITLE
Fix jagged NT softmax semantics

### PR DESCRIPTION
**Pull request by jbschlosser**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #119459

Before: `softmax` definition uses `jagged_unary_pointwise()` (wrong)
After: `softmax` impl adjusts the `dim` arg to account for the difference in dimensionality between the outer NT and the NT's `_values`

### CODE REVIEW

**Review by soulitzer — APPROVED** [review]

**Comment by malfet** [comment]
@pytorchbot rebase 

macos-m1-12 runners are gone long time ago

**Comment by YuqingJ** [comment]
Hi @jbschlosser, any plan to re-land this fix? I plan to add the backward support for softmax, it will be great to fix its forward first. Thanks
